lib/protobuf/field/message_field.rb

```diff
@@ -9,7 +9,7 @@ class MessageField < BaseField
       #
 
       def acceptable?(val)
-        unless val.instance_of?(type_class) || val.respond_to?(:to_hash)
+        unless val.is_a?(type_class) || val.is_a?(Hash)
```

Comment:
this may be a bit outside the scope of this fix.  The reason I made this change was because this [line](https://github.com/localshred/protobuf/pull/252/files#diff-1de145c39faa7b92fc0a33ea3d9fc49bR69) was failing without it.  IT seemed wrong that an incorrect object that has no relationship to the expected object should be acceptable.  It was passing this method because of `val.respond_to?(:to_hash)`, which seems too open ended.  Also, using `instance_of?` doesn't take into account class inheritance while `is_a?` does.